### PR TITLE
Make closed state configurable

### LINKED ISSUE #1592: Make closed state of SHACL shapes configurable
**Is your feature request related to a problem? Please describe.**
linkML generated all SHACL shapes as closed shapes, except of Mixin shapes and shapes, that have a parent class. See [shaclgen.py#L66](https://github.com/linkml/linkml/blob/main/linkml/generators/shaclgen.py#L66). Closed shapes do not allow additional attributes. However, there are uses cases, which want to allow additional attributes in shapes. 

**Describe the solution you'd like**
Extend CLI with an additional optional parameter to configure ```closed``` state of SHACL shapes.

**How important is this feature?** 
• Medium - can do work without it, by e.g. adding a post-generating procedure to remove ```sh:closed true``` from SHACL shapes
• Furthermore making closed state of SHACL shapes is a simple task

**When will use cases depending on this become relevant?**
• Short-term - 2-4 weeks

**Additional context**
I already provided a fix for this feature.

### PR DESCRIPTION
**Pull request by anjastrunk**
closes #1592

### CODE REVIEW

File: tests/test_scripts/test_gen_shacl.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import unittest
+
+from linkml.generators import shaclgen
+from tests.test_scripts.environment import env
+from tests.utils.clicktestcase import ClickTestCase
+
+
+class GenYUMLTestCase(ClickTestCase):
```
Comment (cmungall) [inline tests/test_scripts/test_gen_shacl.py:8]: Rename

**Review by cmungall — COMMENTED** [review]
Thank you!

Normally we require a test case - note we are switching from unittest to pytest (and making it easier to contribute tests). I see you added a CLI base for tests, great. I would recommend that we add a test in test_generators that tests specifically this behavior independent of the CLI.

Let us know if you want help writing a test

**Comment by anjastrunk** [comment]
> Thank you!
> 
> Normally we require a test case - note we are switching from unittest to pytest (and making it easier to contribute tests). I see you added a CLI base for tests, great. I would recommend that we add a test in test_generators that tests specifically this behavior independent of the CLI.
> 
> Let us know if you want help writing a test

I added an additional test case to test configuration of closed parameter. Code coverage still decreased by this PR. However ```shaclgen.py``` (the only class I touched) is now again by 100 %.

**Comment by cmungall** [comment]
@sujaypatil96 can you rebase the branch to get the fixes for codecov?

**Pull request merged**